### PR TITLE
log tuning issues interactively

### PR DESCRIPTION
**Pull request by simonpcouch**
Integrates with https://github.com/tidymodels/tune/pull/593 to enable interactive tuning issue logging.

This PR also transitions a few `rlang::inform()` calls to `cli::cli_bullets()`, the latter of which is aware of the progress bar.

### CODE REVIEW

**Review by DavisVaughan — APPROVED** [review]
Seems reasonable

File: R/tune_race_anova.R [hunk 1]
```diff
@@ -198,6 +198,8 @@ tune_race_anova_workflow <-
            control = control_race()) {
     rlang::check_installed("lme4")
 
+    tune::initialize_catalog(control = control)
```
Comment (topepo) [inline R/tune_race_anova.R:201]: I like that it weaves the messages in-between the logging

```
→ 1 | warning: The specified df was too small.
→ 2 | warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
ℹ Racing will maximize the roc_auc metric. 1: x3   2: x2
ℹ Resamples are analyzed in a random order.
ℹ Fold07: 4 eliminated; 6 candidates remain.             
ℹ Fold09: 0 eliminated; 6 candidates remain.             
→ 3 | warning: Some 'x' values beyond boundary knots may cause ill-conditioned basis
functions.
ℹ Fold10: 0 eliminated; 6 candidates remain.                     
ℹ Fold03: 0 eliminated; 6 candidates remain.                     
ℹ Fold02: 0 eliminated; 6 candidates remain.                     
ℹ Fold08: 0 eliminated; 6 candidates remain.                     
ℹ Fold06: 0 eliminated; 6 candidates remain.                      
There were issues with some computations   1: x10   2: x2   3: x11
```

File: R/tune_sim_anneal.R [hunk 2]
```diff
@@ -354,7 +358,7 @@ tune_sim_anneal_workflow <-
 
     cols <- tune::get_tune_colors()
     if (control$verbose_iter) {
-      rlang::inform(cols$message$info(paste("Optimizing", metrics_name)))
```
Comment (topepo) [inline R/tune_sim_anneal.R:357]: This is a lot to read though: 

```
→ 1 | warning: Some 'x' values beyond boundary knots may cause ill-conditioned basis
functions.
There were issues with some computations   1: x2
Optimizing roc_auc                               
Initial best: 0.88440                            
1 ◯ accept suboptimal  roc_auc=0.88335 (+/-0.01352)
2 + better suboptimal  roc_auc=0.88418 (+/-0.01354)
3 ◯ accept suboptimal  roc_auc=0.88374 (+/-0.01421)
4 ♥ new best  roc_auc=0.88592 (+/-0.01306)        
→ 2 | warning: glm.fit: fitted probabilities numerically 0 or 1 occurred
5 ◯ accept suboptimal  roc_auc=0.88444 (+/-0.01215)       
6 ◯ accept suboptimal  roc_auc=0.88102 (+/-0.01224)       
7 + better suboptimal  roc_auc=0.88473 (+/-0.01238)       
8 ◯ accept suboptimal  roc_auc=0.88385 (+/-0.01252)       
9 ◯ accept suboptimal  roc_auc=0.8836 (+/-0.0132)         
10 + better suboptimal  roc_auc=0.88505 (+/-0.01277)      
There were issues with some computations   1: x22   2: x4
```

The mixing of numbers for iterations and warnings/errors is confusing. 

Would we consider using uppercase letters for the error/warning logging? We could have the pool of labels are `c(LETTERS, letters)` if we think that there will be more than 52 unique cases.

**Comment by topepo** [comment]
Code I used to induce warnings of different types: 

```r

library(tidymodels)
library(finetune)

# ------------------------------------------------------------------------------

tidymodels_prefer()
theme_set(theme_bw())
options(pillar.advice = FALSE, pillar.min_title_chars = Inf)

# ------------------------------------------------------------------------------

data(two_class_dat)

set.seed(1)
rs <- vfold_cv(two_class_dat)

model_wflow <- 
  workflow() %>% 
  add_recipe(
    recipe(Class ~ ., data = two_class_dat) %>% 
      step_spline_b(A, deg_free = tune("a")) %>% 
      step_spline_b(B, deg_free = tune("b"))
  ) %>% 
  add_model(logistic_reg())


set.seed(2)
res_1 <- 
  tune_race_anova(model_wflow, rs, grid = 10)


set.seed(2)
res_2 <-
  tune_race_anova(model_wflow,
                  rs,
                  grid = 10,
                  control = control_race(verbose_elim = TRUE))



set.seed(2)
res_3 <-
  tune_race_anova(model_wflow,
                  rs,
                  grid = 10,
                  control = control_race(verbose = TRUE, verbose_elim = TRUE))



set.seed(2)
res_4 <-
  tune_sim_anneal(model_wflow,
                  rs,
                  iter = 10,
                  control = control_sim_anneal(verbose = TRUE, verbose_iter = TRUE))

```

File: R/tune_sim_anneal.R (hunk 2 above)
Comment (simonpcouch) [reply R/tune_sim_anneal.R:357]: On board! PR at tidymodels/tune#596. :)

**Pull request merged**